test/basic-buy.ts

```diff
@@ -0,0 +1,264 @@
+import { DsProxyLike, IERC20, MPALike } from '../typechain'
+
+const EXCHANGE_ADDRESS = '0xb5eB8cB6cED6b6f8E13bcD502fb489Db4a726C7B'
+const testCdpId = parseInt(process.env.CDP_ID || '13288')
+
+function toRatio(units: number) {
+    return new BigNumber(units).shiftedBy(4).toNumber()
+}
+
+// BLOCK_NUMBER=14997398
+describe('BasicBuyCommand', () => {
+    const ethAIlk = utils.formatBytes32String('ETH-A')
+    const hardhatUtils = new HardhatUtils(hre)
+
+    let system: DeployedSystem
+    let DAIInstance: IERC20
+    let MPAInstance: MPALike
+    let usersProxy: DsProxyLike
+    let proxyOwnerAddress: string
+    let receiverAddress: string
+    let executorAddress: string
+    let snapshotId: string
+
+    before(async () => {
+        executorAddress = await hre.ethers.provider.getSigner(0).getAddress()
+        receiverAddress = await hre.ethers.provider.getSigner(1).getAddress()
+
+        DAIInstance = await hre.ethers.getContractAt('IERC20', hardhatUtils.addresses.DAI)
+        MPAInstance = await hre.ethers.getContractAt('MPALike', hardhatUtils.addresses.MULTIPLY_PROXY_ACTIONS)
+
+        system = await deploySystem({ utils: hardhatUtils, addCommands: true })
+
+        await system.mcdView.approve(executorAddress, true)
+
+        const cdpManager = await hre.ethers.getContractAt('ManagerLike', hardhatUtils.addresses.CDP_MANAGER)
+        const proxyAddress = await cdpManager.owns(testCdpId)
+        usersProxy = await hre.ethers.getContractAt('DsProxyLike', proxyAddress)
+        proxyOwnerAddress = await usersProxy.owner()
+
+        const osmMom = await hre.ethers.getContractAt('OsmMomLike', hardhatUtils.addresses.OSM_MOM)
+        const osm = await hre.ethers.getContractAt('OsmLike', await osmMom.osms(ethAIlk))
+        await hardhatUtils.setBudInOSM(osm.address, system.mcdView.address)
+    })
+
+    beforeEach(async () => {
+        snapshotId = await hre.ethers.provider.send('evm_snapshot', [])
+    })
+
+    afterEach(async () => {
+        await hre.ethers.provider.send('evm_revert', [snapshotId])
+    })
+
+    describe('isTriggerDataValid', () => {
+        const createTrigger = async (triggerData: BytesLike) => {
+            const data = system.automationBot.interface.encodeFunctionData('addTrigger', [
+                testCdpId,
+                TriggerType.BASIC_BUY,
+                0,
+                triggerData,
+            ])
+            const signer = await hardhatUtils.impersonate(proxyOwnerAddress)
+            return usersProxy.connect(signer).execute(system.automationBot.address, data)
+        }
+
+        it('should fail if target coll ratio is higher than execution ratio', async () => {
+            const [executionRatio, targetRatio] = [toRatio(1.01), toRatio(1.02)]
+            const triggerData = encodeTriggerData(
+                testCdpId,
+                TriggerType.BASIC_BUY,
+                executionRatio,
+                targetRatio,
+                0,
+                false,
```

Comment:
do we have any tests with `continous` true?  First check if it do not fails triggers execution,
second if it creates valid trigger with the same trigger data. Arguably third it this other trigger successfully executes

Reply:
a lot of tests are missing, as discussed, this is a draft PR still